myhoard/controller.py

```diff
@@ -1173,7 +1173,7 @@ def _refresh_backups_list(self):
         )
         new_backups_ids = {backup["stream_id"] for backup in backups}
         for backup in self.state["backups"]:
-            if backup["stream_id"] not in new_backups_ids:
+            if backup["stream_id"] not in new_backups_ids and backup["site"] in self.backup_sites:
```

Comment:
I would rewrite as (and add a explanation):
 
```python
for backup in self.state["backups"]:
    if backup["site"] not in self.backup_sites:
        # some explanation why this can happen
        continue
   if backup["stream_id"] not in new_backups_ids:
       self._build_backup_stream(backup).delete_state()
```